vertical-pod-autoscaler/pkg/recommender/input/controller_fetcher/controller_fetcher.go

```diff
@@ -206,6 +209,12 @@ func (f *controllerFetcher) getParentOfController(controllerKey ControllerKeyWit
 		return getParentOfWellKnownController(informer, controllerKey)
 	}
 
+	// check if it's scalable
+	scalable := f.isScalable(controllerKey.ApiVersion, controllerKey.Kind)
+	if !scalable {
```

Comment:
I'm not sure this early exit actually works?
A Pod might have an owner reference that is neither well known nor scalable. As long as the _topmost_ owner (the end of the "ownership chain") is scalable, this should be sufficient. With this change, we would return `nil`, as if this controller didn't have an owner, which doesn't seem correct.

Reply:
What if the topmost owner (the end of the "ownership chain") isn't scalable? Even the spec.TargetRef is scalable, this VPA still won't be synced